packages/core/src/sdk/shipping/usePickupPoints.ts

```diff
@@ -0,0 +1,74 @@
+import { useMemo } from 'react'
+import { gql } from '@generated'
+import type {
+  ClientPickupPointsQueryQuery,
+  ClientPickupPointsQueryQueryVariables,
+} from '@generated/graphql'
+
+import { deliveryPromise } from 'discovery.config'
+import { useSession } from 'src/sdk/session'
+import { useQuery } from 'src/sdk/graphql/useQuery'
+
+export const query = gql(`
+  query ClientPickupPointsQuery(
+    $geoCoordinates: IStoreGeoCoordinates
+    $postalCode: String
+    $country: String
+  ) {
+    pickupPoints(geoCoordinates: $geoCoordinates, postalCode: $postalCode, country: $country) {
+      paging {
+        total
+      }
+      items {
+        pickupPoint {
+          id
+          address {
+            street
+          }
+          friendlyName
+        }
+      }
+    }
+  }
+`)
+
+export const usePickupPoints = () => {
```

Comment:
The important thing here is to observe the number of this query calls (`Network` tab), if it's being made when it shouldn't.